`make lint` failing consistently

## Environment

- **Make target:** `make -C images/capi lint`
- **Run using container image?:** No

## What steps did you take and what happened?

running `make lint` against the `main` branch leads to this error that we are seeing in CI:

```shell
ModuleNotFoundError: No module named 'ansible.parsing.yaml.constructor'
```

## What did you expect to happen?

Linter should run to completion and report no issue.


## Relevant log output

<details>
<summary>Failure Output</summary>

```shell
% make -C images/capi lint                   
hack/ensure-python.sh
Checking if python is available
Python 3.13.5
hack/ensure-ansible.sh
ansible [core 2.19.0]
Starting galaxy collection install process
Nothing to do. All requested collections are already installed. If you want to reinstall them, consider using `--force`.
hack/ensure-packer.sh
Packer is already installed, checking version...
existing packer version: 1.13.1
unsupported packer version (1.13.1) found
current packer version: 1.13.1 is not supported
Downgrading packer to 1.9.5
Installing packer v1.9.5 in .local/bin
packer_1.9.5_darwin_arm64.zip: OK
Archive:  packer_1.9.5_darwin_arm64.zip
  inflating: packer                  
'packer' has been installed to /Users/matt/projects/image-builder/images/capi/.local/bin, make sure this directory is in your $PATH
hack/ensure-ansible-windows.sh
IMPORTANT: Winrm connection plugin for Ansible on MacOS causes connection issues.
See https://docs.ansible.com/ansible/latest/user_guide/windows_winrm.html#what-is-winrm for more details.
To fix the issue provide the enviroment variable 'no_proxy=*'
Example call to build Windows images on MacOS: 'no_proxy=* make build-<target>'
hack/ensure-ansible-lint.sh
hack/ensure-shellcheck.sh
ansible-lint ansible/
Traceback (most recent call last):
  File "/Users/matt/projects/image-builder/env/bin/ansible-lint", line 5, in <module>
    from ansiblelint.__main__ import _run_cli_entrypoint
  File "/Users/matt/projects/image-builder/env/lib/python3.13/site-packages/ansiblelint/__main__.py", line 39, in <module>
    from ansiblelint import cli
  File "/Users/matt/projects/image-builder/env/lib/python3.13/site-packages/ansiblelint/cli.py", line 29, in <module>
    from ansiblelint.yaml_utils import clean_json
  File "/Users/matt/projects/image-builder/env/lib/python3.13/site-packages/ansiblelint/yaml_utils.py", line 33, in <module>
    from ansiblelint.utils import Task
  File "/Users/matt/projects/image-builder/env/lib/python3.13/site-packages/ansiblelint/utils.py", line 43, in <module>
    from ansible.parsing.yaml.constructor import AnsibleConstructor, AnsibleMapping
ModuleNotFoundError: No module named 'ansible.parsing.yaml.constructor'
make: *** [lint] Error 1
```
</details>




---

/kind bug

--- a/images/capi/hack/ensure-ansible-lint.sh
+++ b/images/capi/hack/ensure-ansible-lint.sh
@@ -32,6 +32,6 @@ cd "$(dirname "${BASH_SOURCE[0]}")/.."
 export PIP_DISABLE_PIP_VERSION_CHECK=1 PIP_ROOT_USER_ACTION=ignore
 
 if ! command -v ansible-lint >/dev/null 2>&1; then
-    pip3_install "ansible-lint==${_version}"
+    pip3_install "ansible-lint==${_version}" "ansible-core==${_version_ansible_core}"
     ensure_py3_bin ansible-lint
 fi

--- a/images/capi/hack/ensure-ansible.sh
+++ b/images/capi/hack/ensure-ansible.sh
@@ -22,9 +22,6 @@ set -o pipefail
 
 source hack/utils.sh
 
-# Note: ansible-core v2.16.x requires Python >= 3.10.
-_version="2.15.13"
-
 # Change directories to the parent directory of the one in which this
 # script is located.
 cd "$(dirname "${BASH_SOURCE[0]}")/.."
@@ -33,18 +30,18 @@ cd "$(dirname "${BASH_SOURCE[0]}")/.."
 export PIP_DISABLE_PIP_VERSION_CHECK=1 PIP_ROOT_USER_ACTION=ignore
 
 if ! command -v ansible >/dev/null 2>&1; then
-    pip3_install "ansible-core==${_version}"
+    pip3_install "ansible-core==${_version_ansible_core}"
     ensure_py3_bin ansible
     ensure_py3_bin ansible-playbook
 fi
 
 ansible_version=""
 IFS=" " read -ra ansible_version <<< "$(ansible --version)"
-if [[ "${_version}" != $(echo -e "${_version}\n${ansible_version[2]}" | sort -s -t. -k 1,1 -k 2,2n -k 3,3n | head -n1) && "${ansible_version[2]}" != "devel" ]]; then
+if [[ "${_version_ansible_core}" != $(echo -e "${_version_ansible_core}\n${ansible_version[2]}" | sort -s -t. -k 1,1 -k 2,2n -k 3,3n | head -n1) && "${ansible_version[2]}" != "devel" ]]; then
   cat <<EOF
 Detected ansible version: ${ansible_version[*]}.
-Image builder requires ${_version} or greater.
-Please install ${_version} or later.
+Image builder requires ${_version_ansible_core} or greater.
+Please install ${_version_ansible_core} or later.
 EOF
   exit 2
 fi

--- a/images/capi/hack/utils.sh
+++ b/images/capi/hack/utils.sh
@@ -14,6 +14,9 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# Note: ansible-core v2.16.x requires Python >= 3.10.
+_version_ansible_core="2.15.13"
+
 case "${OSTYPE}" in
 linux*)
   HOSTOS=linux

